Makefile

```diff
@@ -1,184 +1,103 @@
-	make elastica-image
-	make start
-	mkdir -p build
-
-	docker run -e "ES_HOST=elasticsearch" --network=elastica_esnet elastica_elastica make phpunit
-	docker cp elastica:/elastica/build/coverage/ $(shell pwd)/build/coverage
-
-# Makes it easy to run a single test file. Example to run IndexTest.php: make test TEST="IndexTest.php"
-.PHONY: test
-test:
-	make elastica-image
-	make start
-	mkdir -p build
-	docker-compose run -e "ES_HOST=elasticsearch" elastica phpunit -c test/ ${TEST}
-
-.PHONY: doc
-doc:
-	${RUN_ENV} phpdoc run -d lib/ -t build/docs --template=/root/composer/vendor/phpdocumentor/phpdocumentor/data/templates/clean
-
-# Uses the preconfigured standards in .php_cs.dist
-.PHONY: lint
-lint:
-	docker build -t ruflin/elastica-dev-base-linter -f env/elastica/Lint env/elastica/
-	${RUN_ENV_LINT} php-cs-fixer fix --allow-risky=yes
-
-.PHONY: check-style
-check-style:
-	docker build -t ruflin/elastica-dev-base-linter -f env/elastica/Lint env/elastica/
-	${RUN_ENV_LINT} php-cs-fixer fix --allow-risky=yes --dry-run --diff
-
-.PHONY: loc
-loc:
-	${RUN_ENV} cloc --by-file --xml --exclude-dir=build -out=build/cloc.xml .
-
-.PHONY: phploc
-phploc:
-	${RUN_ENV} phploc --log-csv ./build/logs/phploc.csv ${SOURCE}
-
-
-# VIRTUAL ENVIRONMENT
-.PHONY: build
-build:
-	docker-compose build
-
-.PHONY: start
-start:
-	docker-compose up -d --build
-
-.PHONY: stop
-stop:
-	docker-compose stop
-
-.PHONY: destroy
-destroy: clean
-	docker-compose kill
-	docker-compose rm
-
-# Stops and removes all containers and removes all images
-.PHONY: destroy-environment
-destroy-environment:
-	make remove-containers
-	-docker rmi $(shell docker images -q)
-
-.PHONY: remove-containers
-remove-containers:
-	-docker stop $(shell docker ps -a -q)
-	-docker rm -v $(shell docker ps -a -q)
-
-# Starts a shell inside the elastica image
-.PHONY: shell
-shell:
-	docker run -v $(shell pwd):/elastica -ti ruflin/elastica /bin/bash
-
-# Starts a shell inside the elastica image with the full environment running
-.PHONY: env-shell
-env-shell:
-	docker-compose run elastica /bin/bash
-
-# Visualise repo
+.PHONY: run-coveralls
+run-coveralls:
+	tools/php-coveralls.phar -v
+
+tools/phpdocumentor.phar:
+	curl https://gitreleases.dev/gh/phpDocumentor/phpDocumentor/latest/phpDocumentor.phar -o tools/phpdocumentor.phar --silent -L; \
+	chmod +x tools/phpdocumentor.phar
+
+.PHONY: run-phpdoc
+run-phpdoc: tools/phpdocumentor.phar
+	tools/phpdocumentor.phar --directory=lib --target=build/docs --template=clean
+
+##
+## Docker commands
+##
+
+.PHONY: docker-start
+docker-start:
+	sudo sysctl -w vm.max_map_count=262144; \
+	docker-compose --file=docker/docker-compose.yml --file=docker/docker-compose.es.yml --file=docker/docker-compose.es${START_ES_VERSION}.yml up --build
+
+.PHONY: docker-stop
+docker-stop:
+	docker-compose --file=docker/docker-compose.yml --file=docker/docker-compose.es.yml --file=docker/docker-compose.es${START_ES_VERSION}.yml down
+
+.PHONY: docker-run-phpunit
+docker-run-phpunit:
```

Comment:
I documented it in the CONTRIBUTING: as a developer you should keep an eye to the ES logs, right? :)
My solution: run `make docker-start` in one terminal, and run `make docker-run-phpunit` in another. The ES logs will be visible in the first terminal, including errors and logs in case one of the containers fail to start.

In this was the following error could be found and fixed in any local environment ;-)
```
es02_1  | ERROR: [1] bootstrap checks failed
es02_1  | [1]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
```